test/unit/specs/components/DecisioningEngine/decisioningContext.referringPage.spec.js

```diff
@@ -0,0 +1,373 @@
+import createOnResponseHandler from "../../../../../src/components/DecisioningEngine/createOnResponseHandler";
+import createDecisionProvider from "../../../../../src/components/DecisioningEngine/createDecisionProvider";
+import createContextProvider from "../../../../../src/components/DecisioningEngine/createContextProvider";
+import createEventRegistry from "../../../../../src/components/DecisioningEngine/createEventRegistry";
+
+const mockWindow = ({
```

Comment:
Rather than duplicate these helper methods in each test file, please extract the mockWindow, payloadWithCondition, mockRulesetResponseWithCondition, setupResponseHandler methods to a separate context test utils file that each of these tests use.